classlib/src/main/java/org/teavm/classlib/java/util/TTreeMap.java

```diff
@@ -80,11 +80,11 @@ public void fix() {
         }
 
         public TreeNode<K, V> forward(boolean reverse) {
-            return !reverse ? left : right;
+            return reverse ? right : left;
```

Comment:
Not sure why you did this commit. I added inversion here intentially, since I believe that this code is *more* readable, when *normal* condition comes first.

Reply:
Anyway, having bugfix and style commit together in one PR makes it difficult to review such PR

Reply:
For me notinverted looks more readable. I'm assuming default value is false and looking to right when checking what value it should be. Also in other places it allowed to remove double inversions.

Reply:
I suppose that you can restore inverses after merging this PR in places you consider most important for readability.

Reply:
I don't agree with your point. The very meaning of this method is following. Consider we want to find next node in a binary tree. Then we move right and after that step move left until it's possible. Searching for the previous node is symmetric one: first move left and then move right until it's possible. Since these are symmetric, it was reasonable to generalize this algorithm. So what's a way to do this? Say that we always find "next" node, but there's option to search in forward or backward direction. For me, it's natural that we first formulate task as "forward" direction case, since it's easier to model in a head. So this "forward" case moves first. On the other hand, how should we call a flag that determines direction? For me it's natural to have "false" value for something by default, so there's no "forward" flag, but "reverse". This gives all these inversions. Perhaps, it's reasonable to have `forward(): boolean` method that simply inverts `reverse` flag.

There could be also concern that direction should not be encoded with boolean, rather with enum. However, reference to enum value takes 4 or 8 bytes, while `boolean` may occupy only one byte. Performance is quite important in library code.

Reply:
> I suppose that you can restore inverses after merging this PR in places you consider most important for readability.

No, why would I add extra garbage to git log? In case you refuse to revert that commit, I won't merge this PR, rather add my commits to this branch and them make squash

Reply:
Reverted.